### PR TITLE
task/DES-1194: Minimal Python 3 Upgrade

### PR DESCRIPTION
**Pull request by jarosenb**
For local work in the Django shell, use `/home/tg458981/portal_env/bin/python manage.py...`

### CODE REVIEW

File: conf/docker/Dockerfile [hunk 1]
```diff
@@ -1,49 +1,85 @@
-FROM buildpack-deps:trusty
+FROM centos:7
```
Comment (rstijerina) [inline conf/docker/Dockerfile]: thoughts on keeping a debian based image, but updating to something like `python:3.6-buster`?

File: designsafe/middleware.py [hunk 2]
```diff
@@ -67,18 +67,16 @@ def process_request(self, request):
...
-                                         '/terms/accept/') + term
-                    messages.warning(
-                        request, '<h4>Please Accept the Terms of Use</h4>'
-                                 'You have not yet agreed to the current Terms of Use. '
-                                 'Please <a href="%s">CLICK HERE</a> to review and '
-                                 'accept the Terms of Use.<br>Acceptance of the Terms of '
-                                 'Use is required for continued use of DesignSafe '
-                                 'resources.' % accept_url)
-                    #return redirect(reverse('designsafe_accounts:profile_edit'))
+            if TermsAndConditions.get_active_terms_not_agreed_to(request.user):
```
[the remark this answers is not in the document]
Comment (jarosenb) [reply designsafe/middleware.py]: Yeah, the python3-compatible version of this library has a different API.

File: conf/docker/Dockerfile (hunk 1 above)
Comment (jarosenb) [reply conf/docker/Dockerfile]: I'll look into it, most of the dockerfile came from Josue's original Py3 PR.

File: designsafe/apps/api/agave/filemanager/lookups.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 """
    :synopsis: Function to lookup manager classes
 """
-from __future__ import unicode_literals, absolute_import
```
[the remark this answers is not in the document]
Comment (jarosenb) [reply designsafe/apps/api/agave/filemanager/lookups.py]: I don't think there are implications for a Django version bump if Python3 is being run before and after. They got removed by 2to3 but I'll drop them back in if we run into issues during testing.

**Comment by rstijerina** [comment]
@jarosenb will you be merging in your debian changes here? They appear to be working

**Pull request closed without merging**